AMM/contracts/AMM/tests/utils/mod.rs

```diff
@@ -0,0 +1,152 @@
+use fuels::{contract::contract::CallResponse, prelude::*};
+use rand::prelude::Rng;
+use std::str::FromStr;
+
+abigen!(AMM, "out/debug/amm-abi.json");
+abigen!(Exchange, "../exchange/out/debug/exchange-abi.json");
+
+pub mod exchange_abi_calls {
+    use super::*;
+
+    pub async fn constructor(
+        contract: &Exchange,
+        pair: (ContractId, ContractId),
+    ) -> CallResponse<()> {
+        let receipt = contract.methods().constructor(pair).call().await;
+        dbg!(&receipt);
+        receipt.unwrap()
+    }
+}
+
+pub mod amm_abi_calls {
+    use super::*;
+
+    pub async fn initialize(contract: &AMM, exchange_contract_id: ContractId) -> CallResponse<()> {
+        contract
+            .methods()
+            .initialize(exchange_contract_id)
+            .call()
+            .await
+            .unwrap()
+    }
+
+    pub async fn add_pool(
+        contract: &AMM,
+        asset_pair: (ContractId, ContractId),
+        pool: ContractId,
+    ) -> CallResponse<()> {
+        contract
+            .methods()
+            .add_pool(asset_pair, pool)
+            .set_contracts(&[pool.into()])
+            .call()
+            .await
+            .unwrap()
+    }
+
+    pub async fn pool(contract: &AMM, asset_pair: (ContractId, ContractId)) -> Option<ContractId> {
+        contract
+            .methods()
+            .pool(asset_pair)
+            .call()
+            .await
+            .unwrap()
+            .value
+    }
+}
+
+pub mod paths {
```

Comment:
I opened 1 issue per app except the [multisig wallet](https://github.com/FuelLabs/sway-applications/blob/master/multisig-wallet/tests/harness.rs), since it is heavily WIP.